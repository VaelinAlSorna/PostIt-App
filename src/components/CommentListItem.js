import React from 'react';

class CommentListItem extends React.Component {
   
    onDelete = () => {
        this.props.onClickDeleteComment(this.props.id);
    }
    render () { 
        return  (
                <li>
                    <p>{this.props.createdAt}</p>
                    <p>{this.props.comment}</p>
                    <button onClick={this.onDelete}>Delete</button>
                </li>
            )
    }
};

export default CommentListItem;