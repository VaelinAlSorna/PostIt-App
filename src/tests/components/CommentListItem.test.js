import React from 'react';
import { shallow } from 'enzyme';
import CommentListItem from '../../components/CommentListItem';
import articles from '../fixtures/articlesFixtures';

test('should render CommentListItem correctly', () => {
    const wrapper = shallow(<CommentListItem {...articles[0].comments[0]} />);
    expect(wrapper).toMatchSnapshot();
});

test('should handle delete button onClick', () => {
    const onClickDeleteComment = jest.fn();
    const wrapper = shallow(<CommentListItem 
                                onClickDeleteComment={onClickDeleteComment} 
                                {...articles[0].comments[0]}
                            />
                        );
    wrapper.find('button').simulate('click');
    expect(onClickDeleteComment).toHaveBeenCalledWith(articles[0].comments[0].id);

});