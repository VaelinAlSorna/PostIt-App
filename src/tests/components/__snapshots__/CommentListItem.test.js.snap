// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CommentListItem correctly 1`] = `
<li>
  <p>
    1.1.1970
  </p>
  <p>
    Something to test?
  </p>
  <button
    onClick={[Function]}
  >
    Delete
  </button>
</li>
`;
